xbrowse_server/base/models.py

```diff
@@ -192,19 +194,19 @@ def can_view(self, user):
 
         if self.is_public:
             return True
-        elif user.is_staff:
+        elif (user.is_staff and not self.disable_staff_access) or ProjectCollaborator.objects.filter(project=self, user=user).exists():
             return True
         else:
             return ProjectCollaborator.objects.filter(project=self, user=user).exists()
```

Comment:
you will check `ProjectCollaborators` here twice

Reply:
you're right.